noxfile-template.py

```diff
@@ -113,8 +181,53 @@ def py(session):
     if session.python in TESTED_VERSIONS:
         _session_tests(session)
     else:
-        print("SKIPPED: {} tests are disabled for this sample.".format(session.python))
+        session.skip("SKIPPED: {} tests are disabled for this sample.".format(
+            session.python
+        ))
+
+
+#
+# cloud_run session
+#
+
+@nox.session
+def cloud_run(session):
+    """Run tests for cloud run."""
+    if 'cloud_run' not in TEST_CONFIG['opt_in_sessions']:
+        session.skip('SKIPPED: cloud_run tests are disabled for this sample.')
+        return
+
+    if os.path.exists("requirements.txt"):
+        session.install("-r", "requirements.txt")
+
+    if os.path.exists("requirements-test.txt"):
+        session.install("-r", "requirements-test.txt")
 
+    user_envs = get_pytest_env_vars()
+
+    # Only update gcloud on Kokoro.
+    if os.environ.get("KOKORO_JOB_NAME"):
+        # Activate service account
+        key_file = os.environ['GOOGLE_APPLICATION_CREDENTIALS']
+        session.run(
+            "gcloud", "auth", "activate-service-account",
+            "--key-file={}".format(key_file)
+        )
```

Comment:
The auth library should be handling this already - we shouldn't need to do this as a part of the test

Reply:
It's for gcloud command itself, not for the library usage. The current Cloud Run sample author prefers to use gcloud for deploying the app.

Reply:
Apologies, I misunderstood how this was being used. I thought this was being used to set the service account used in the Run instance, but it's actually being used to authenticate the glcoud command to deploy the instance.

In this case, it would be best to handle this in our test script, instead of in the noxfile. The `gcloud auth` makes changes to the local gcloud config, meaning that running this test has side effects and might effect the other tests. If we set this [here](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/2e502b1da42d5de8db1de3e49c6a020e87714c67/.kokoro/tests/run_tests.sh#L45) instead, it's consistent for all of the tests.

Reply:
Ok, I can do that.